dao/src/main/java/org/thingsboard/server/dao/entity/BaseEntityService.java

```diff
@@ -125,100 +74,34 @@ public PageData<EntityData> findEntityDataByQuery(TenantId tenantId, CustomerId
-            case ASSET:
-                hasName = assetService.findAssetByIdAsync(tenantId, new AssetId(entityId.getId()));
-                break;
-            case DEVICE:
-                hasName = deviceService.findDeviceByIdAsync(tenantId, new DeviceId(entityId.getId()));
-                break;
-            case ENTITY_VIEW:
-                hasName = entityViewService.findEntityViewByIdAsync(tenantId, new EntityViewId(entityId.getId()));
-                break;
-            case TENANT:
-                hasName = tenantService.findTenantByIdAsync(tenantId, TenantId.fromUUID(entityId.getId()));
-                break;
-            case CUSTOMER:
-                hasName = customerService.findCustomerByIdAsync(tenantId, new CustomerId(entityId.getId()));
-                break;
-            case USER:
-                hasName = userService.findUserByIdAsync(tenantId, new UserId(entityId.getId()));
-                break;
-            case DASHBOARD:
-                hasName = dashboardService.findDashboardInfoByIdAsync(tenantId, new DashboardId(entityId.getId()));
-                break;
-            case ALARM:
-                hasName = alarmService.findAlarmByIdAsync(tenantId, new AlarmId(entityId.getId()));
-                break;
-            case RULE_CHAIN:
-                hasName = ruleChainService.findRuleChainByIdAsync(tenantId, new RuleChainId(entityId.getId()));
-                break;
-            case EDGE:
-                hasName = edgeService.findEdgeByIdAsync(tenantId, new EdgeId(entityId.getId()));
-                break;
-            case TB_RESOURCE:
-                hasName = resourceService.findResourceInfoByIdAsync(tenantId, new TbResourceId(entityId.getId()));
-                break;
-            case OTA_PACKAGE:
-                hasName = otaPackageService.findOtaPackageInfoByIdAsync(tenantId, new OtaPackageId(entityId.getId()));
-                break;
-            default:
-                throw new IllegalStateException("Not Implemented!");
+    public Optional<String> fetchEntityName(TenantId tenantId, EntityId entityId) {
+        log.trace("Executing fetchEntityName [{}]", entityId);
+        TbEntityService tbEntityService = entityServiceBeanFactory.getServiceByEntityType(entityId.getEntityType());
+        Optional<HasId<?>> hasIdOpt = tbEntityService.fetchEntity(tenantId, entityId);
+        if (hasIdOpt.isPresent()) {
+            HasId<?> hasId = hasIdOpt.get();
+            if (hasId instanceof HasName) {
+                HasName hasName = (HasName) hasId;
+                return Optional.ofNullable(hasName.getName());
+            }
         }
-        entityName = Futures.transform(hasName, (Function<HasName, String>) hasName1 -> hasName1 != null ? hasName1.getName() : null, MoreExecutors.directExecutor());
-        return entityName;
+        return Optional.empty();
     }
 
     @Override
     public CustomerId fetchEntityCustomerId(TenantId tenantId, EntityId entityId) {
```

Comment:
This method should also return the Optional<CustomerId> to be consistent with the `fetchEntityName`